src/compositions/hooks.ts

```diff
@@ -48,7 +48,7 @@ function afterComponentHookFactory(lifecycle: AfterRouteHookLifecycle) {
  * @returns {RouteHookRemove} A function that removes the added hook.
  * @group Hooks
  */
-export const onBeforeRouteLeave: AddBeforeRouteHook = beforeComponentHookFactory('onBeforeRouteUpdate')
+export const onBeforeRouteLeave: RegisterBeforeRouteHook = beforeComponentHookFactory('onBeforeRouteUpdate')
```

Comment:
There was a mix of `Register` and `Add` keywords used in hooks. Tried to unify the types around `Register`.

Reply:
Actually the types were register and the methods were add. So I renamed all the types to use "Add" to match the methods.